scripts/ci-e2e.sh

```diff
@@ -84,6 +84,25 @@ fi
 
 "hack/ci/create_devstack.sh"
 
+# TODO: Upload image for e2e clusterctl upgrade tests
+# This is just a quick hack to test the process
+source "${REPO_ROOT}/hack/ci/${RESOURCE_TYPE}.sh"
+CONTAINER_ARCHIVE="${ARTIFACTS}/capo-e2e-image.tar"
+SSH_KEY="$(get_ssh_private_key_file)"
+SSH_ARGS="-o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o PasswordAuthentication=no"
+CONTROLLER_IP=${CONTROLLER_IP:-"10.0.3.15"}
+
+make e2e-image
+docker save -o "${CONTAINER_ARCHIVE}" gcr.io/k8s-staging-capi-openstack/capi-openstack-controller:e2e
+scp -i "${SSH_KEY}" ${SSH_ARGS} "${CONTAINER_ARCHIVE}" "cloud@${CONTROLLER_IP}:capo-e2e-image.tar"
+ssh -i "${SSH_KEY}" ${SSH_ARGS} "cloud@${CONTROLLER_IP}" -- sudo chown root:root capo-e2e-image.tar
+ssh -i "${SSH_KEY}" ${SSH_ARGS} "cloud@${CONTROLLER_IP}" -- sudo chmod u=rw,g=r,o=r capo-e2e-image.tar
+ssh -i "${SSH_KEY}" ${SSH_ARGS} "cloud@${CONTROLLER_IP}" -- sudo mv capo-e2e-image.tar /var/www/html/capo-e2e-image.tar
+
+# TODO: This is just to test the clusterctl upgrade tests
+export E2E_GINKGO_PARALLEL="1"
+export E2E_GINKGO_ARGS="-focus='\[clusterctl-upgrade\]'"
```

Comment:
The idea with this would be to pass it in directly in test-infra, similar to how it is [done for CAPI](https://github.com/kubernetes/test-infra/blob/245873efc4d80d737143d7ebf62d5493495cba0a/config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml#L140-L141).